src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -300,30 +300,44 @@ private string InsertDeletedRange(DocumentsOperationContext context, DeletionRan
                 (changeVector, etag) = GenerateChangeVector(context);
             }
 
-            using (var sliceHolder = new TimeSeriesSliceHolder(context, documentId, name, collectionName.Name).WithEtag(etag))
-            using (table.Allocate(out var tvb))
-            using (Slice.From(context.Allocator, changeVector, out var cv))
+            var hash = (long)Hashing.XXHash64.Calculate(changeVector, Encoding.UTF8);
+
+            using (var sliceHolder = new TimeSeriesSliceHolder(context, documentId, name, collectionName.Name).WithChangeVectorHash(hash))
             {
-                tvb.Add(sliceHolder.TimeSeriesKeySlice);
-                tvb.Add(Bits.SwapBytes(etag));
-                tvb.Add(cv);
-                tvb.Add(sliceHolder.CollectionSlice);
-                tvb.Add(context.GetTransactionMarker());
-                tvb.Add(from.Ticks);
-                tvb.Add(to.Ticks);
+                if (table.ReadByKey(sliceHolder.TimeSeriesKeySlice, out var tableValueReader))
+                {
+                    var item = CreateDeletedRangeItem(context, ref tableValueReader);
+                    if (ChangeVectorUtils.GetConflictStatus(changeVector, item.ChangeVector) == ConflictStatus.AlreadyMerged)
```

Comment:
actually you need this check only if you have a hash collision, otherwise you can simply check whether this exists in the table or not.
But let's say you do have a hash collision and the conflict status is `Conflicted`, what then?
I would argue it is so rare that we don't care and can simply check if the value exists or not.

Reply:
in that case, a hash collision will overwrite the existing one :-)

Reply:
ok